### PR TITLE
OWM Vacancy Manager scenarios: SSJ application

### PR DESCRIPTION
**Pull request by alenaminsk**
OWM Vacancy Manager scenarios: SSJ application

### CODE REVIEW

File: pom.xml
Comment (iamdez99) [inline pom.xml:1]: Remove runners name from pom

File: src/main/java/com/nci/automation/web/PlaywrightUtils.java
Comment (iamdez99) [inline src/main/java/com/nci/automation/web/PlaywrightUtils.java:1]: Can you please change setHeadless() from true to false? It currently is false on master. I will work on making this dynamic in my next PR.

File: src/main/resources/conf/env/test.xml
Comment (iamdez99) [inline src/main/resources/conf/env/test.xml:1]: is the '#/' needed at the end of the URL?

File: src/test/java/ServiceNow/PlatformBusinessApps/SSJ/playwright/Steps/ApplicantProfileSteps.java
Comment (iamdez99) [inline src/test/java/ServiceNow/PlatformBusinessApps/SSJ/playwright/Steps/ApplicantProfileSteps.java:1]: There is a commented line of code on this file.

File: src/test/java/ServiceNow/PlatformBusinessApps/SSJ/playwright/StepsImplementation/ApplicantProfileStepsImpl.java
Comment (iamdez99) [inline src/test/java/ServiceNow/PlatformBusinessApps/SSJ/playwright/StepsImplementation/ApplicantProfileStepsImpl.java:1]: Empty line with imports.. please check

File: src/test/java/ServiceNow/PlatformBusinessApps/SSJ/playwright/StepsImplementation/Reset_Account_StepsImpl.java
Comment (iamdez99) [inline src/test/java/ServiceNow/PlatformBusinessApps/SSJ/playwright/StepsImplementation/Reset_Account_StepsImpl.java:1]: Where is the EncryptionUtils import being used?

File: src/test/java/appsCommon/PlaywrightUtils/Playwright_ServiceNow_Common_Methods.java
Comment (iamdez99) [inline src/test/java/appsCommon/PlaywrightUtils/Playwright_ServiceNow_Common_Methods.java:1]: Please review your PRs with more attention to detail. This is the whole point of QA's reviewing PRs before asking the team to review so we dont point out simple errors.

**Review by iamdez99 — CHANGES_REQUESTED** [review]

File: pom.xml
Comment (alenaminsk) [reply pom.xml:1]: removed

File: src/main/java/com/nci/automation/web/PlaywrightUtils.java
Comment (alenaminsk) [reply src/main/java/com/nci/automation/web/PlaywrightUtils.java:1]: changed

File: src/main/resources/conf/env/test.xml
Comment (alenaminsk) [reply src/main/resources/conf/env/test.xml:1]: removed

File: src/test/java/ServiceNow/PlatformBusinessApps/SSJ/playwright/StepsImplementation/ApplicantProfileStepsImpl.java
Comment (alenaminsk) [reply src/test/java/ServiceNow/PlatformBusinessApps/SSJ/playwright/StepsImplementation/ApplicantProfileStepsImpl.java:1]: removed

File: src/test/java/ServiceNow/PlatformBusinessApps/SSJ/playwright/StepsImplementation/Reset_Account_StepsImpl.java
Comment (alenaminsk) [reply src/test/java/ServiceNow/PlatformBusinessApps/SSJ/playwright/StepsImplementation/Reset_Account_StepsImpl.java:1]: removed

File: src/test/java/appsCommon/PlaywrightUtils/Playwright_ServiceNow_Common_Methods.java
Comment (alenaminsk) [reply src/test/java/appsCommon/PlaywrightUtils/Playwright_ServiceNow_Common_Methods.java:1]: Thank you Diego

File: src/test/java/ServiceNow/PlatformBusinessApps/SSJ/playwright/Features/Applicant Profile.feature [hunk 1]
```diff
@@ -31,5 +31,29 @@ Feature: Applicant Profile Scenarios
...
+      | Violeta   | Seidell    | Lack      | seidellvv@nih.gov  | 4133212333 | 2902122234    | Bachelors        | Yes             | 2 Bells  | 2313      | Monterey  | CA    | USA     | 94333 |
+      | Abby      | Marianna   | Molnar    | molnaraj@nih.gov   | 7018211343 | 2123323454    | Masters          | Yes             | 11 Kolls | 6         | Carmel    | CA    | USA     | 97812 |
+      | Maureen   | Jim        | Kennedy   | mkennedyh@nih.gov  | 8233212333 | 2202122234    | Bachelors        | Yes             | 11 Bells | 39        | San Jose  | CA    | USA     | 95112 |
+      | Jessica   | Marianna   | Mollick   | mollickja@nih.gov  | 7018211343 | 2123323454    | Masters          | Yes             | 11 Kolls | 6         | Sausalito | CA    | USA     | 97712 |
+      | Karen     | Elena      | Andrew    | andrewkl@nih.gov   | 8233212333 | 2202122234    | Bachelors        | Yes             | 11 Bells | 39        | Sacramento| CA    | USA     | 95512 |
+
+  @Regression @Smoke @alenan @APPTRACK-712
```
Comment (hamidnesar) [inline src/test/java/ServiceNow/PlatformBusinessApps/SSJ/playwright/Features/Applicant Profile.feature:46]: Is SSJ and APPTRACK the same application? If not, the story tag should be adjusted accordingly

File: src/test/java/ServiceNow/PlatformBusinessApps/SSJ/playwright/Features/OWM_Vacancy_Manager.feature [hunk 2]
```diff
@@ -0,0 +1,236 @@
+Feature: OWM Vacancy Manager Scenarios
+  Description: This Feature file contains OWM Vacancy Manager Scenarios
+
+  @SSJ-7  @SSJ-8
```
Comment (hamidnesar) [inline src/test/java/ServiceNow/PlatformBusinessApps/SSJ/playwright/Features/OWM_Vacancy_Manager.feature:4]: Missing @regression or @smoke tags

File: src/test/java/ServiceNow/PlatformBusinessApps/SSJ/playwright/Features/OWM_Vacancy_Manager.feature [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+    And User verifies Number of Scoring Categories slider has 6 options
+    And User verifies that all positions are present via Position Classification dropdown
+    And  User verifies all codes are present via Organizational Code dropdown
+    And User verifies the text of "Position Classification" i icon
+    And User verifies the text of "Organizational Code" i icon
+    And User verifies the text of "Personnel Action Tracking Solution (PATS) Initiator" i icon
+    And User clicks "Save" button
+    And User confirms that User is on "Mandatory Statements" section
+
+  @SSJ-69
```
Comment (hamidnesar) [inline src/test/java/ServiceNow/PlatformBusinessApps/SSJ/playwright/Features/OWM_Vacancy_Manager.feature:36]: Missing @regression or @smoke tags

File: src/test/java/ServiceNow/PlatformBusinessApps/SSJ/playwright/Features/OWM_Vacancy_Manager.feature [hunk 4]
```diff
@@ -0,0 +1,236 @@
...
+    And User verifies the text of "Reasonable Accommodation" is displayed as "NIH provides reasonable accommodations to applicants with disabilities. If you require reasonable accommodation during any part of the application and hiring process, please notify us. The decision on granting reasonable accommodation will be made on a case-by-case basis."
+    And User can disable statement options
+      | Standards of Conduct/Financial Disclosure | Foreign Education | Reasonable Accommodation |
+    And can enable statement options
+      | Standards of Conduct/Financial Disclosure | Foreign Education | Reasonable Accommodation |
+    And User clicks "Save" button
+    And User confirms that User is on "Vacancy Committee" section
+
+  @SSJ-185 @SSJ-306
```
Comment (hamidnesar) [inline src/test/java/ServiceNow/PlatformBusinessApps/SSJ/playwright/Features/OWM_Vacancy_Manager.feature:56]: Missing @regression or @smoke tags

File: src/test/java/ServiceNow/PlatformBusinessApps/SSJ/playwright/Features/OWM_Vacancy_Manager.feature [hunk 5]
```diff
@@ -0,0 +1,236 @@
...
+    And User adds a Committee Chair
+    And User adds a Member (voting)
+    And User adds a Member (non-voting)
+    And User adds a HR Specialist (non-voting)
+    And User adds EDI Representative (non-voting)
+    And User adds Executive Secretary (non-voting)
+    And User clicks "Save" section button
+    And User confirms that User is on "Email Templates" section
+
+  @SSJ-11
```
Comment (hamidnesar) [inline src/test/java/ServiceNow/PlatformBusinessApps/SSJ/playwright/Features/OWM_Vacancy_Manager.feature:81]: Missing @regression or @smoke tags

File: src/test/java/ServiceNow/PlatformBusinessApps/SSJ/playwright/Features/OWM_Vacancy_Manager.feature [hunk 6]
```diff
@@ -0,0 +1,236 @@
...
+    And User clicks on "Vacancy Committee" option on the left bar menu
+    When User clicks button "Add Member"
+    And User adds a Committee Chair
+    Then User sees alert "At least one committee member must be of the role 'Executive Secretary"
+    And User removes a Commitee Chair
+    Then User sees alert "At least one committee member must be of the role 'Chair' At least one committee member must be of the role 'Executive Secretary"
+    And User adds Executive Secretary (non-voting)
+    Then User sees alert "At least one committee member must be of the role 'Chair'"
+
+  @SSJ-310 @SSJ-158 @SSJ-71
```
Comment (hamidnesar) [inline src/test/java/ServiceNow/PlatformBusinessApps/SSJ/playwright/Features/OWM_Vacancy_Manager.feature:97]: Missing @regression or @smoke tags

File: src/test/java/ServiceNow/PlatformBusinessApps/SSJ/playwright/Features/OWM_Vacancy_Manager.feature [hunk 7]
```diff
@@ -0,0 +1,236 @@
...
+    And User clicks "Save" section button
+    And User verifies that "Email Templates"  section title is displayed
+    And User confirms that "Application saved" and "Application submitted confirmation" templates are enabled by default
+    And User can disable "Application saved" and "Application submitted confirmation" templates choice
+    And User verifies the text of "Application saved" template
+    And User verifies the tect of "Application submitted confirmation" template
+    And User clicks "Save" section button
+    And User verifies that "Review and Finalize" section title is displayed
+
+  @SSJ-64  @SSJ-305
```
Comment (hamidnesar) [inline src/test/java/ServiceNow/PlatformBusinessApps/SSJ/playwright/Features/OWM_Vacancy_Manager.feature:116]: Missing @regression or @smoke tags

File: src/test/java/ServiceNow/PlatformBusinessApps/SSJ/playwright/Features/OWM_Vacancy_Manager.feature [hunk 8]
```diff
@@ -0,0 +1,236 @@
...
+    And User clicks Edit for "Basic Vacancy Information" section
+    And User sees "Please enter a description." alert for Vacancy Description fields
+    And User sees "Please select an open date" alert for Open Date entry
+    And User sees "Please make a selection" for Position Classification field
+    And User sees "Please make a selection" for Organizational Code field
+    And User clicks on Review and Finalize left menu option
+    And User clicks Edit for "Vacancy Committee" section
+    Then User can verify alert "At least one committee member must be of the role 'Chair' At least one committee member must be of the role 'Executive Secretary'" is present
+
+  @SSJ-221 @SSJ-100
```
Comment (hamidnesar) [inline src/test/java/ServiceNow/PlatformBusinessApps/SSJ/playwright/Features/OWM_Vacancy_Manager.feature:145]: Missing @regression or @smoke tags

File: src/test/java/ServiceNow/PlatformBusinessApps/SSJ/playwright/Features/OWM_Vacancy_Manager.feature [hunk 9]
```diff
@@ -0,0 +1,236 @@
...
+    And User adds a Committee Chair
+    And User adds Executive Secretary (non-voting)
+    And User verifies that "Email Templates"  section title is displayed
+    And User clicks "Save" section button
+    When User clicks "Save and Finalize" button
+    And User can see a new live vacancy on a Vacancy Dashboard under Live Vacancies
+    When User navigates to Home tab
+    Then User can see a new live vacancy under Open Vacancies
+
+  @SSJ-1  @SSJ-29 @SSJ-66 @SSJ-62
```
Comment (hamidnesar) [inline src/test/java/ServiceNow/PlatformBusinessApps/SSJ/playwright/Features/OWM_Vacancy_Manager.feature:170]: Missing @regression or @smoke tags

File: src/test/java/ServiceNow/PlatformBusinessApps/SSJ/playwright/Features/OWM_Vacancy_Manager.feature [hunk 10]
```diff
@@ -0,0 +1,236 @@
...
+      | All | Live | Extended |
+    And User sees live vacancies colums
+      | Vacancy Title | Open Date | Close Date | Actions |
+    When User is on closed vacancies
+    Then User sees the following closed vacancies filters
+      | AlL | Triage | Individual | ScoringCommittee | ReviewVoting | Complete |
+    And User sees closed vacancies colums
+      | Vacancy Title | Open Date | Close Date | Actions |
+
+  @SSJ-160 @SSJ-284
```
Comment (hamidnesar) [inline src/test/java/ServiceNow/PlatformBusinessApps/SSJ/playwright/Features/OWM_Vacancy_Manager.feature:193]: Missing @regression or @smoke tags

File: src/test/java/ServiceNow/PlatformBusinessApps/SSJ/playwright/Features/OWM_Vacancy_Manager.feature [hunk 11]
```diff
@@ -0,0 +1,236 @@
...
+    And User closes the opened  new web page
+    When User clicks Extend
+    Then User sees Extend Vacancy modal appear
+    And User sets a date
+    And User clicks Confirm
+    Then User can verify the updated close date
+    And User navigates to Home tab
+    Then User can verify the updated close date on Home Tab for that live vacancy
+
+  @SSJ-93 @SSJ-146
```
Comment (hamidnesar) [inline src/test/java/ServiceNow/PlatformBusinessApps/SSJ/playwright/Features/OWM_Vacancy_Manager.feature:219]: Missing @regression or @smoke tags

File: src/test/java/ServiceNow/PlatformBusinessApps/SSJ/playwright/Pages/Profile_Tab_Page.java [hunk 12]
```diff
@@ -77,8 +77,53 @@ public class Profile_Tab_Page {
...
 
+    /**
+     * Demographics (optional) tab
+     */
+    public static String demographicsOptionalTab = "//div[@id='rc-tabs-0-tab-demographics']";
+
+    /**
+     * Demographics (optional) Edit Button
+     */
+    public static String demographicsEditButton = "//span[.='Edit']";
```
Comment (hamidnesar) [inline src/test/java/ServiceNow/PlatformBusinessApps/SSJ/playwright/Pages/Profile_Tab_Page.java:88]: On lines 88 and 103, is the period (.) necessary in //span[**.**=

**Review by hamidnesar — CHANGES_REQUESTED** [review]
Please review the comments. Thanks!

File: src/test/java/ServiceNow/PlatformBusinessApps/SSJ/playwright/Features/Applicant Profile.feature (hunk 1 above)
Comment (alenaminsk) [reply src/test/java/ServiceNow/PlatformBusinessApps/SSJ/playwright/Features/Applicant Profile.feature:46]: Adjusted

File: src/test/java/ServiceNow/PlatformBusinessApps/SSJ/playwright/Features/OWM_Vacancy_Manager.feature (hunk 2 above)
Comment (alenaminsk) [reply src/test/java/ServiceNow/PlatformBusinessApps/SSJ/playwright/Features/OWM_Vacancy_Manager.feature:4]: These are only feature files, once glue code is developed, @regression tag will be added

File: src/test/java/ServiceNow/PlatformBusinessApps/SSJ/playwright/Features/OWM_Vacancy_Manager.feature (hunk 3 above)
Comment (alenaminsk) [reply src/test/java/ServiceNow/PlatformBusinessApps/SSJ/playwright/Features/OWM_Vacancy_Manager.feature:36]: @regression tag will be added after glue code is developed

File: src/test/java/ServiceNow/PlatformBusinessApps/SSJ/playwright/Features/OWM_Vacancy_Manager.feature (hunk 5 above)
Comment (alenaminsk) [reply src/test/java/ServiceNow/PlatformBusinessApps/SSJ/playwright/Features/OWM_Vacancy_Manager.feature:81]: @regression tag will be added after glue code is developed

File: src/test/java/ServiceNow/PlatformBusinessApps/SSJ/playwright/Features/OWM_Vacancy_Manager.feature (hunk 6 above)
Comment (alenaminsk) [reply src/test/java/ServiceNow/PlatformBusinessApps/SSJ/playwright/Features/OWM_Vacancy_Manager.feature:97]: @regression tag will be added after glue code is developed

File: src/test/java/ServiceNow/PlatformBusinessApps/SSJ/playwright/Features/OWM_Vacancy_Manager.feature (hunk 10 above)
Comment (alenaminsk) [reply src/test/java/ServiceNow/PlatformBusinessApps/SSJ/playwright/Features/OWM_Vacancy_Manager.feature:193]: @regression tag will be added after glue code is developed

File: src/test/java/ServiceNow/PlatformBusinessApps/SSJ/playwright/Features/OWM_Vacancy_Manager.feature (hunk 9 above)
Comment (alenaminsk) [reply src/test/java/ServiceNow/PlatformBusinessApps/SSJ/playwright/Features/OWM_Vacancy_Manager.feature:170]: @regression tag will be added after glue code is developed

File: src/test/java/ServiceNow/PlatformBusinessApps/SSJ/playwright/Features/OWM_Vacancy_Manager.feature (hunk 11 above)
Comment (alenaminsk) [reply src/test/java/ServiceNow/PlatformBusinessApps/SSJ/playwright/Features/OWM_Vacancy_Manager.feature:219]: @regression tag will be added after glue code is developed

File: src/test/java/ServiceNow/PlatformBusinessApps/SSJ/playwright/Pages/Profile_Tab_Page.java (hunk 12 above)
Comment (alenaminsk) [reply src/test/java/ServiceNow/PlatformBusinessApps/SSJ/playwright/Pages/Profile_Tab_Page.java:88]: Yes, (.) is a part of a locator

**Review by iamdez99 — CHANGES_REQUESTED** [review]
Please review changes

File: src/test/java/ServiceNow/PlatformBusinessApps/SSJ/playwright/Features/Applicant Profile.feature
Comment (matakevin) [inline src/test/java/ServiceNow/PlatformBusinessApps/SSJ/playwright/Features/Applicant Profile.feature:1]: Remove the extra space after Diego's tag on line 4

File: src/test/java/ServiceNow/PlatformBusinessApps/SSJ/playwright/Features/OWM_Vacancy_Manager.feature
Comment (matakevin) [inline src/test/java/ServiceNow/PlatformBusinessApps/SSJ/playwright/Features/OWM_Vacancy_Manager.feature:1]: Are these scenarios in progress, a part of smoke or regression?  Please add tags symbolizing state of the scenarios

File: src/test/java/ServiceNow/PlatformBusinessApps/SSJ/playwright/Steps/ApplicantProfileSteps.java
Comment (matakevin) [inline src/test/java/ServiceNow/PlatformBusinessApps/SSJ/playwright/Steps/ApplicantProfileSteps.java:1]: There's a couple of steps here that have more than 1 line of code, can you ask Diego if that's okay or if you can add the double clicks in one method or the screenshot line of code also

File: src/test/java/ServiceNow/PlatformBusinessApps/SSJ/playwright/StepsImplementation/ApplicantProfileStepsImpl.java
Comment (matakevin) [inline src/test/java/ServiceNow/PlatformBusinessApps/SSJ/playwright/StepsImplementation/ApplicantProfileStepsImpl.java:1]: Are there any actions being take for code on lines 21 and 23?

**Review by matakevin — CHANGES_REQUESTED** [review]
Left some comments, please review

File: src/test/java/ServiceNow/PlatformBusinessApps/SSJ/playwright/Features/Applicant Profile.feature
Comment (alenaminsk) [reply src/test/java/ServiceNow/PlatformBusinessApps/SSJ/playwright/Features/Applicant Profile.feature:1]: removed

File: src/test/java/ServiceNow/PlatformBusinessApps/SSJ/playwright/Features/OWM_Vacancy_Manager.feature
Comment (alenaminsk) [reply src/test/java/ServiceNow/PlatformBusinessApps/SSJ/playwright/Features/OWM_Vacancy_Manager.feature:1]: updated

File: src/test/java/ServiceNow/PlatformBusinessApps/SSJ/playwright/Steps/ApplicantProfileSteps.java
Comment (alenaminsk) [reply src/test/java/ServiceNow/PlatformBusinessApps/SSJ/playwright/Steps/ApplicantProfileSteps.java:1]: updated

File: src/test/java/ServiceNow/PlatformBusinessApps/SSJ/playwright/StepsImplementation/ApplicantProfileStepsImpl.java
[the remark this answers is not in the document]
Comment (alenaminsk) [reply src/test/java/ServiceNow/PlatformBusinessApps/SSJ/playwright/StepsImplementation/ApplicantProfileStepsImpl.java:1]: removed

Comment (alenaminsk) [reply src/test/java/ServiceNow/PlatformBusinessApps/SSJ/playwright/StepsImplementation/ApplicantProfileStepsImpl.java:1]: to be developed later

**Review by iamdez99 — APPROVED** [review]
Reviewed and approved

**Pull request merged**